pkg/telemetry/tracing.go

```diff
@@ -5,43 +5,77 @@ import (
 	"fmt"
 	"time"
 
-	"github.com/openfga/openfga/internal/build"
 	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/codes"
 	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
 	"go.opentelemetry.io/otel/propagation"
 	"go.opentelemetry.io/otel/sdk/resource"
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
-	semconv "go.opentelemetry.io/otel/semconv/v1.12.0"
 	"go.opentelemetry.io/otel/trace"
 	"google.golang.org/grpc"
 )
 
-func MustNewTracerProvider(endpoint string, serviceName string, ratio float64) *sdktrace.TracerProvider {
+type TracerOption func(d *CustomTracer)
+
+func WithOTLPEndpoint(endpoint string) TracerOption {
+	return func(d *CustomTracer) {
+		d.endpoint = endpoint
+	}
+}
+
+func WithSamplingRatio(samplingRatio float64) TracerOption {
+	return func(d *CustomTracer) {
+		d.samplingRatio = samplingRatio
+	}
+}
+
+func WithAttributes(attrs ...attribute.KeyValue) TracerOption {
+	return func(d *CustomTracer) {
+		d.attributes = attrs
+	}
+}
+
+type CustomTracer struct {
```

Comment:
Does this need to be exported? If we're going to have a construct return the `TracerProvider`, then I'd recommend we unexport this, because the intent is to build a contract on the OTEL interfaces, not the CustomTracer directly.

Reply:
No need to be exported